ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -438,7 +441,7 @@ deleteValue
   -> NormalizedFilePath
   -> IO ()
 deleteValue ShakeExtras{dirtyKeys, state} key file = do
-    void $ modifyVar' state $ HMap.delete (toKey key file)
+    atomically $ STM.delete (toKey key file) state
     atomicModifyIORef_ dirtyKeys $ HSet.insert (toKey key file)
```

Comment:
This pairing seems a bit suspicious! Maybe these two modifications *should* be in a transaction together? I guess we can't fix that without changing `dirtyKeys`.

Reply:
Yeah, these should be mutated atomically. It's not hard to change `dirtyKeys`, I'll add it to this PR